### PR TITLE
Increase generator validation strictness and associated tests

### LINKED ISSUE #150: Validate against blank/null fields in the generator

### PR DESCRIPTION
**Pull request by Kapin**
Closes #150

### CODE REVIEW

File: src-backend/api/generator.py [hunk 1]
```diff
@@ -8,6 +8,12 @@ def __init__(self, procedure):
         self.name = 'Procedure'
         self.procedure = procedure
 
+        if not self.procedure.title:
```
Comment (Ommy) [inline src-backend/api/generator.py]: is this testing if self.procedure.title is none? If so, why not use `if self.procedure.title is None:`

File: src-backend/api/generator.py [hunk 2]
```diff
@@ -43,6 +52,24 @@ def __init__(self, element):
         self.name = 'Element'
         self.element = element
 
+        if not self.element.element_type:
+            self.__raise_error('Element with id %d has no type')
```
Comment (Ommy) [inline src-backend/api/generator.py]: All these `%d`s don't have any variable reference.

Comment (Kapin) [inline src-backend/api/generator.py]: it does. It smacks in the element id inside __raise_error

**Comment by Trinovantes** [comment]
+1

**Comment by Ommy** [comment]
LGTM

**Pull request merged**